Add Points of Interest backend API: /otp/routers/default/pois

jmfield2 opened the pull request:
Filtering of returned points supported via "q" request parameter.

Map Builds now support build-config.json:

{
 pois: { amenity: {
                     parking: [],
                     bicycle_repair_station: [],
                     bicycle_parking: [],
                     charging_station: [],
                     fast_food: [],
                     restaurant: []
                  },
         shop:    {
                     supermarket: []
                  },
         sport: {
                    soccer: [],
                    baseball: [],
                    football: [],
                    softball: [],
                    basketball: [],
                    tennis: [],
                    running: []
                }
        }
}

And frontend UI configured using js/otp/config.js

layersWidget: {}
     { name: 'parking', target: '#parkinglots', icon: new L.layersIcon({icon: 'mdi mdi-parking', color: 'red'}), type: 'poi', search: 'amenity:parking', popupTemplate: 'parkingLots-template', condition: "permits", color: "red" },

```
Target is DOM selector defined in the client, typically in the layer widget template: usf-layer-menu in layers/layers-templates.html

Templates stored in js/otp/layers/layers-templates.html
Search matches key=value based on backend API "q" param
```

Route E color, and bike lanes updated fixes #204

Add links for EV stations for #227

Also for #177, #179

jmfield2 commented:
So, there are a handful of whitespace-related lines in this merge I think .. nothing major, but there.

Also, @jailby I think I regressed the 'box_col' class in layers-templates.html  @ https://github.com/CUTR-at-USF/usf-mobullity/pull/241/files#diff-bf2eecadf70c25f97650281de7bcfad1R125

jailby commented on src/client/js/otp/widgets/tripoptions/LayersWidget.js:
```diff
@@ -64,142 +64,166 @@ otp.widgets.LayersWidget =
-        
-	// Bullrunner bus stops+routes
+
+	    // Bullrunner bus stops+routes
```
This one

jailby commented:
Yeah I saw the white space too.
I think it got read of the useless whitespace except one that I commented.

Yeah I see that you are not using `box_col` anymore. Can you briefly explain me why you did so? Was it the issue where the text needed to be on two lines?

jmfield2 commented:
No reason, I think I just used my changes over upstream during the merge
which is probably wrong .. I forgot if you changed that or not so I can
change them back to how they are on mobullity
On Mar 27, 2016 9:07 PM, "jailby" notifications@github.com wrote:

> Yeah I saw the white space too.
> I think it got read of the useless whitespace except one that I commented.
> 
> Yeah I see that you are not using box_col anymore. Can you briefly
> explain me why you did so? Was it the issue where the text needed to be on
> two lines?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/CUTR-at-USF/usf-mobullity/pull/241#issuecomment-202180044

jailby commented:
hmm I do not remember changing that so I don't think it was me.

jmfield2 committed:
Add Points of Interest backend API: /otp/routers/default/pois

Filtering of returned points supported via "q" request parameter.

Map Builds now support build-config.json:

{
 pois: { amenity: {
                     parking: [],
                     bicycle_repair_station: [],
                     bicycle_parking: [],
                     charging_station: [],
                     fast_food: [],
                     restaurant: []
                  },
         shop:    {
                     supermarket: []
                  },
         sport: {
                    soccer: [],
                    baseball: [],
                    football: [],
                    softball: [],
                    basketball: [],
                    tennis: [],
                    running: []
                }
        }
}

And frontend UI configured using js/otp/config.js

layersWidget: {}
     { name: 'parking', target: '#parkinglots', icon: new L.layersIcon({icon: 'mdi mdi-parking', color: 'red'}), type: 'poi', search: 'amenity:parking', popupTemplate: 'parkingLots-template', condition: "permits", color: "red" },

    Target is DOM selector defined in the client, typically in the layer widget template: usf-layer-menu in layers/layers-templates.html

    Templates stored in js/otp/layers/layers-templates.html
    Search matches key=value based on backend API "q" param

Route E color, and bike lanes updated for #204

Add links for EV stations for #227
